### PR TITLE
feat(adshboard,types,medusa): enable adding notes/return reason to inbound claims

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds toast messages to failed actions
- add ability to create and update notes / return reason
- fixes types in claims components

RESOLVES CC-297

https://github.com/user-attachments/assets/a3a14384-49e5-4d20-a9ef-5379b1122740

### CODE REVIEW

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx [hunk 1]
```diff
@@ -161,14 +166,18 @@ export const ClaimCreateForm = ({
...
-        inbound_items: previewItems.map((i) => ({
-          item_id: i.id,
-          quantity: i.detail.return_requested_quantity,
-          note: i.actions?.find((a) => a.action === "RETURN_ITEM")
-            ?.internal_note,
-          reason_id: i.actions?.find((a) => a.action === "RETURN_ITEM")?.details
-            ?.reason_id,
-        })),
+        inbound_items: previewItems.map((i) => {
+          const returnAction = i.actions?.find(
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:170]: thought: Should we check if we found a `returnAction` here before sending it to the backend, or does it not matter?

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx [hunk 2]
```diff
@@ -341,6 +368,10 @@ export const ClaimCreateForm = ({
       )
         .filter((it) => it?.variant)
         .forEach((item) => {
+          if (!item) {
+            return
+          }
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx]:
```suggestion
        .filter((it) => !!it?.variant)
        .forEach((item) => {
```

**Review by olivermrbl — APPROVED** [review]
Nice work!

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx (hunk 1 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:170]: The return action will always be present here. Right now in the previewItems array, there are only inbound items as we can only create those atm. When I get to creating other types of actions, I will have to change the logic of the previewItems here and extract inboundItems from there. At that point, this logic here will be removed.

Comment (olivermrbl) [reply packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:170]: Got it, thx 👍

**Pull request merged**